src/main/java/edu/kis/vh/nursery/list/IntLinkedList.java

```diff
@@ -2,39 +2,67 @@
 
 public class IntLinkedList {
 
-    Node lastElement;
-    int i;
+    final int EMPTY_LIST_VALUE = -1;
 
-    public void push(int valueToAdd) {
+    private int value;
+    private IntLinkedList lastElement;
```

Comment:
Moving Node responsibility to IntLinkedList is an error! 
Node and original IntLinkedList should be reverted. Node can be moved inside IntLinkedList.